### PR TITLE
Modular computer startup program

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

Adds a var to the modular computer to specify a default startup program.

## Why It's Good For The Game

Moving it from a cargo computer specific proc to a general modular computer one allows any preset modular computer to have a startup program.

## Changelog

:cl: LT3
code: All preset modular computers can now have a program launch automatically on startup
/:cl:

### CODE REVIEW

File: code/modules/modular_computers/computers/machinery/console_presets.dm [hunk 1]
```diff
@@ -74,43 +91,41 @@
...
+	preinstalled_programs = list(
 		/datum/computer_file/program/portrait_printer,
 	)
 
 // ===== CARGO CHAT CONSOLES =====
 /obj/machinery/modular_computer/preset/cargochat
 	name = "cargo interfacing console"
 	desc = "A stationary computer that comes pre-loaded with software to interface with the cargo department."
-	starting_programs = list(
+	startup_program = "dept_order"
```
Comment (MrMelbert) [inline code/modules/modular_computers/computers/machinery/console_presets.dm:102]: As a thought: These should be
`/datum/computer_file/program/department_order::filename`
, etc. 
So if their filename is updated these don't get left behind mistakenly

File: code/modules/modular_computers/computers/machinery/console_presets.dm [hunk 2]
```diff
@@ -74,43 +91,41 @@
...
+	preinstalled_programs = list(
 		/datum/computer_file/program/chatclient,
+		/datum/computer_file/program/department_order
```
Comment (MrMelbert) [inline code/modules/modular_computers/computers/machinery/console_presets.dm:105]:
```suggestion
		/datum/computer_file/program/department_order,
```

File: code/modules/modular_computers/computers/machinery/console_presets.dm [hunk 3]
```diff
@@ -132,18 +147,15 @@
...
-	update_appearance(UPDATE_ICON)
-	// Rest of the chat program setup is done in LateInit
+	startup_program = "ntnrc_client"
+	preinstalled_programs = list(
+		/datum/computer_file/program/chatclient,
+		/datum/computer_file/program/department_order,
+		/datum/computer_file/program/bounty_board,
+		/datum/computer_file/program/budgetorders,
+		/datum/computer_file/program/shipping,
+		/datum/computer_file/program/restock_tracker
```
Comment (MrMelbert) [inline code/modules/modular_computers/computers/machinery/console_presets.dm:157]:
```suggestion
		/datum/computer_file/program/restock_tracker,
```

File: code/modules/modular_computers/computers/machinery/console_presets.dm (hunk 1 above)
Comment (lessthnthree) [reply code/modules/modular_computers/computers/machinery/console_presets.dm:102]: Yeah, I was torn about specifying a datum or filename based on how the proc loads the file, this suggestion is better.

**Comment by ShizCalev** [comment]
@lessthnthree failing all unit tests.

**Comment by lessthnthree** [comment]
Yeaaah I haven't taken time to ask about this one.

**Pull request closed without merging**